testutils/test_server.go

```diff
@@ -294,6 +294,9 @@ func (ts *TestServer) addChannel(createChannel func(t testing.TB, opts *ChannelO
 func (ts *TestServer) close(ch *tchannel.Channel) {
 	ch.Close()
 	ts.waitForChannelClose(ch)
```

Comment:
can we replace the `waitForChannelClose` with a select statement on the channel instead? The `select` should have a timeout as well.

we should remove all existing loops where we wait for close with the channel instead